atd-vzd/data_model_readme.md

```diff
@@ -0,0 +1,369 @@
+
+# import units into db.cris_units
+$ psql postgres://visionzero:visionzero@localhost:5432/atd_vz_data -c "SET client_min_messages TO WARNING;" -c "\copy db.cris_units(unit_id, crash_id, unit_type_id)  FROM 'units.csv' HEADER CSV;"
+```
+
+#### Validation queries
+
+Throughout the testing, you can use these `select` queries to observe the results. We're going to be working with crash ID `9999999`. 
+
+```sql
+-- the original CRIS data (each row is a unit with a crash joined to it)
+select
+    cris_units.*,
+    unit_types.description as unit_type_description,
+    cris_crashes.*,
+    road_types.description as road_type_description
+from
+    cris_units
+    left join cris_crashes on cris_units.crash_id = cris_crashes.crash_id
+    left join road_types on cris_crashes.road_type_id = road_types.id
+    left join unit_types on cris_units.unit_type_id = unit_types.id
+    where cris_units.crash_id = 9999999;
+
+-- the VZ data which holds edits (each row is a unit with a crash joined to it)
+select
+    vz_units.*,
+    unit_types.description as unit_type_description,
+    vz_crashes.*,
+    road_types.description as road_type_description
+from
+    vz_units
+    left join vz_crashes on vz_units.crash_id = vz_crashes.crash_id
+    left join road_types on vz_crashes.road_type_id = road_types.id
+    left join unit_types on vz_units.unit_type_id = unit_types.id
+    where vz_units.crash_id = 9999999;
+
+-- the "truth" which includes the calculated `location_id` column (each row is a unit with a crash joined to it)
+select
+    units.*,
+    unit_types.description as unit_type_description,
+    crashes.*,
+    road_types.description as road_type_description
+from
+    units
+    left join crashes on units.crash_id = crashes.crash_id
+    left join road_types on crashes.road_type_id = road_types.id
+    left join unit_types on units.unit_type_id = unit_types.id
+    where units.crash_id = 9999999;
+```
+
+### Tests 👨‍🔬
+
+0. In order to see the debugging messages, set the message level to NOTICE
+
+```sql
+SET client_min_messages TO NOTICE;
+```
+
+#### 1. CRIS user creates a crash record with two unit records. 
+
+```sql
+insert into cris_crashes (
+    crash_id, primary_address, road_type_id, latitude, longitude
+) values (9999999, '1 Fake St', 1, 30.2800238, -97.743370);
+
+insert into cris_units (unit_id, crash_id, unit_type_id) values (987654321, 9999999, 1), (987654322, 9999999, 1);
+```
+
+#### 2. VZ user changes a crash’s Location ID by updating the crash lat/lon
+
+```sql
+update vz_crashes set latitude = 30.36444128, longitude = -97.72865645 where crash_id = 9999999;
+```
+
+Observe that the `location_id` of the `crashes` record changes to `AA7729AE83`.
+
+#### 3. VZ user edits a unit type
+
+```sql
+update vz_units set unit_type_id = 3 where unit_id = 987654321;
+```
+
+#### 4. CRIS user updates the crash lat/lon and road type
+
+```sql
+update cris_crashes set latitude = 1, longitude = 1, road_type_id = 5 where crash_id = 9999999;
```

Comment:
i was confused at first bc this was printing "Crash latitude and longitude have not changed. No location update needed" but realized thats cuz the location trigger was firing and printing that even though no updates were actually being made to the crashes table.. just something that caught my eye, could avoid this by moving the logic for whether to fire the location function into the trigger creation like i mentioned in the other comment.

also has me thinking about whether its odd that technically the crashes table is getting updated here even when no values are actually changing? not a big deal though and fixing that may be more trouble than its worth... would require longer triggers that check to see if theres actually anything to update before executing the update statement but ya just thinkin out loud here! :thinking:

Reply:
so helpful! i started [a slack thread](https://austininnovation.slack.com/archives/C05D9MKS5J9/p1710515515387689) about this.